hardhat.config.js

```diff
@@ -13,8 +14,8 @@ module.exports = {
   networks: {
     hardhat: {
       forking: {
-        // forking is enabled only if FORKING_URL env is provided
-        enabled: !!process.env.FORKING_URL,
+        // forking is enabled only if FORKING_BLOCK is provided (i.e. resetFork was called)
+        enabled: !!process.env.FORKING_BLOCK,
```

Comment:
What is the reason for changing this to `FORKING_BLOCK`? In all the system tests we set the forking block inside the code. So far it was enough to `FORKING_URL=<url> yarn test:system` without providing `FORKING_BLOCK`.

Reply:
We used to run unit tests and system tests separately, so we could rely on `FORKING_URL` to indicate that we want to enable forking or not (we passed `FORKING_URL` for `system tests` and didn't pass it for `unit tests`).

However, to generate coverage reports we need to run **both** unit tests and system tests in one command (some of our contracts are tested in system tests only). So when executing the command, `FORKING_URL` needs to be known (because system tests need it). That's why we cannot rely on whether it is passed or not (it's always passed).

But I've noticed that `FORKING_BLOCK` is only set for system tests (when function `resetFork` is used) and I used it to distinguish between unit tests and system tests.

Also note that if we enabled forking for unit tests, they would run extremely slow (even though `resetFork` is not used in them, the field `enabled` set to `true` is enough). I tried that and they ran 15-20 min or more.

Reply:
I think it boils down to a question if we can fix coverage for system tests. When running coverage for system tests only the result table is completely wrong, shows 100% coverage only for the interfaces and some stubs. There is no notion of coverage of the actual code, even though we test it. After some experiments, it looks like the problem relies in `resetFork()` and `hardhat_reset` method. The problem is known and someone asked very similar question on [gitter](https://gitter.im/sc-forks/solidity-coverage?at=5fdd99ecce40bd3cdbecc5dd) (Pls search for "Quick question, I'm running solidity coverage with hardhat.", cause I can't link exactly his question). The same person suggested to use `evm_snapshot` and `evm_revert` instead.
So if we can fix coverage for system tests, then we can decide whether we should leave or change to what you've proposed. Same for my [comment](https://github.com/keep-network/coverage-pools/pull/128#discussion_r661497775) below.